test/functional/test_framework/util.py

```diff
@@ -181,21 +181,37 @@ def initialize_datadir(dirname, n):
     datadir = os.path.join(dirname, "node"+str(n))
     if not os.path.isdir(datadir):
         os.makedirs(datadir)
-    rpc_u, rpc_p = rpc_auth_pair(n)
     with open(os.path.join(datadir, "bitcoin.conf"), 'w', encoding='utf8') as f:
         f.write("regtest=1\n")
-        f.write("rpcuser=" + rpc_u + "\n")
-        f.write("rpcpassword=" + rpc_p + "\n")
         f.write("port="+str(p2p_port(n))+"\n")
         f.write("rpcport="+str(rpc_port(n))+"\n")
         f.write("listenonion=0\n")
     return datadir
-
-def rpc_auth_pair(n):
-    return 'rpcuser💻' + str(n), 'rpcpass🔑' + str(n)
-
-def rpc_url(i, rpchost=None):
-    rpc_u, rpc_p = rpc_auth_pair(i)
+    
+def get_datadir_path(dirname, n):
+    return os.path.join(dirname, "node"+str(n))
+    
+def get_auth_cookie(datadir, n):
+    if os.path.isfile(os.path.join(datadir, "regtest", ".cookie")):
+        with open(os.path.join(datadir, "regtest", ".cookie"), 'r') as f:
+            userpass = f.read()
+            split_userpass = userpass.split(':')
+            return split_userpass[0], split_userpass[1]
+    else:
+        with open(os.path.join(datadir, "bitcoin.conf"), 'r') as f:
+            user = None
+            password = None
+            for line in f:
+                if line.startswith("rpcuser="):
+                    user = line.split("=")[1].strip("\n")
```

Comment:
Perhaps assert if there are multiple "rpcuser=" lines? Same for "rpcpassword=" lines